### ISSUE TITLE
Token authentication

### ISSUE THREAD

**Issue by dorianim** [bauerj/paperless_app#81]
**Is your feature request related to a problem? Please describe.**
I am using the `PAPERLESS_ENABLE_HTTP_REMOTE_USER` authentication. Therefore, I cannot log in with username and password.

**Describe the solution you'd like**
It would be nice, to be able to use a token for authentication in this case.

**Describe alternatives you've considered**
-

**Additional context**
As far as I can tell, Django token authentication is already enabled in Paperless-ng. Tokens can be created in the [Admin settings](https://paperless.dorian.im/admin/authtoken/tokenproxy/). 
The corresponding Django docs are here: https://www.django-rest-framework.org/api-guide/authentication/#tokenauthentication

**Comment by pree** [comment]
I'm doing something similar, with an oauth2-proxy in front of paperless-ngx. To get the app running I've exposed /api without the proxy, which made it possible to login.

PDF's are however not downloaded, as the paperless app still uses the old `/fetch/doc/xx` endpoint instead of the newer `/api/document/<pk>/download` [Ref](https://paperless-ngx.readthedocs.io/en/latest/api.html#downloading-documents)

**Comment by bauerj** [comment]
Yes, token authentication should be implemented although I planned on requesting a token though the app after a basic-Auth login. I assume that wouldn't help in your case though.

@pree Thanks for pointing that out. This should be changed in the next release. Could you please open a separate issue to change the URL?

**Comment by pree** [comment]
> @pree Thanks for pointing that out. This should be changed in the next release. Could you please open a separate issue to change the URL?

Done #82 :)

**Comment by dkowis** [comment]
I have just set up paperless-ngx behind an oauth2 proxy as well, and sadly I can't authenticate using the token. 

There is a [pull request in progress](https://github.com/paperless-ngx/paperless-ngx/pull/1746) to add oidc auth to it, so it'd be super handy to havea way to put in a token.
